engine/unity5/Assets/Scripts/Robot/RobotBase.cs

```diff
@@ -139,17 +158,29 @@ public bool InitializeRobot(string directory)
             List<RigidNode_Base> nodes = new List<RigidNode_Base>();
             RootNode = BXDJSkeleton.ReadSkeleton(directory + "\\skeleton.bxdj");
             RootNode.ListAllNodes(nodes);
-            
-                foreach (RigidNode_Base Base in RootNode.ListAllNodes())
-                {
+
+            emuList = new List<EmuNetworkInfo>();
+
+            foreach (RigidNode_Base Base in RootNode.ListAllNodes())
+            {
                 try
                 {
                     if (Base.GetSkeletalJoint().attachedSensors != null)
                     {
-                        foreach(RobotSensor sensor in Base.GetSkeletalJoint().attachedSensors)
+                        foreach (RobotSensor sensor in Base.GetSkeletalJoint().attachedSensors)
                         {
-                            Debug.Log(sensor.type.ToString() + " " + sensor.conTypePort1 + " " + sensor.conTypePort2 + " " + sensor.conversionFactor + 
+                            Debug.Log(sensor.type.ToString() + " " + sensor.conTypePort1 + " " + sensor.conTypePort2 + " " + sensor.conversionFactor +
```

Comment:
Might want to remove this eventually to reduce log bloat, i somewhat remember this for-each loop not being called so we should look into that